source/gmsh/utilities.cc

```diff
@@ -54,11 +54,13 @@ namespace Gmsh
                                            const AdditionalParameters &prm)
   {
     std::string base_name = prm.output_base_name;
-    FILE *tmp_file = nullptr;
+    char dir_template[] = "ctfbc-XXXXXX";
     if (base_name == "")
       {
-        tmp_file = std::tmpfile();
-        AssertThrow(tmp_file != nullptr, ExcMessage("Creating temporary file failed!"));
+        const char * temp = mkdtemp(dir_template);
```

Comment:
I have to admit that I am still not a huge fan of using posix `mkdtemp` here. For the simple reason that it is not portable.

I am happy with merging this to master and the release branch at the moment, though. But I suggest we bite the bullet and finally start using `boost::filesystem` to do creation of temporary files/directories and file path handling uniformly and properly.

Reply:
unfortunately, we do not bundle the necessary boost headers to use this, IIRC.

Reply:
we should probably do that so that we can start using `std_cxx17::filesystem` and friends.

Reply:
But please let's wait with this until after the release :-)